nixos/modules/services/networking/znc/default.nix

```diff
@@ -255,7 +270,12 @@ in
         User = cfg.user;
         Group = cfg.group;
         Restart = "always";
-        ExecStart = "${pkgs.znc}/bin/znc --foreground --datadir ${cfg.dataDir} ${escapeShellArgs cfg.extraFlags}";
+        ExecStart = ''
+          ${optionalString cfg.proxychains.enable "${pkgs.proxychains}/bin/proxychains4 "}${pkgs.znc}/bin/znc \
```

Comment:
I don't really like how non-generic this is. Feels like this should be configurable independently somehow, though I don't really know how that would look

Reply:
If we have N different proxy services and M different services, using this pattern means that we need N*M NixOS options to support all combinations

Reply:
It's certainly achievable with overlay or setting a wrapper via `systemd.services...` but this makes it easy to just enable the support.

I'm not concerned about N*M explosion since this is the only commonly used proxy(/wrapper) with ZNC.

Tangentially related this also needs setting up at least

```nix
programs.proxychains.enable = true;
services.tor.client.enable = true;
```

If you prefer I can drop this from the PR and carry that downstream.

Reply:
> If we have N different proxy services and M different services, using this pattern means that we need N*M NixOS options to support all combinations

I support @Infinisil's opinion. This PR is good overall and should be merged IMO.